scipy/_lib/_array_api.py

```diff
@@ -260,6 +261,17 @@ def default_xp(xp: ModuleType) -> Generator[None, None, None]:
         _default_xp_ctxvar.reset(token)
 
 
+def eager_warns(x, warning_type, match=None):
```

Comment:
If we're going to see this a lot, I'd prefer a signature closer to that of `pytest.warns` (and all of our `xp_` functions), like
```suggestion
def eager_warns(warning_type, match=None, xp=xp):
```

I suppose the reason that [`is_lazy_array`](https://github.com/data-apis/array-api-compat/blob/5e14b53a3558765a8f9b921c72f0249cc0c1c5b9/array_api_compat/common/_helpers.py#L959) accepts an array (rather than having a function `is_lazy(xp)` that accepts a namespace) is that an unknown backend could in principle have both lazy and eager arrays? But I'm not sure that is important to us right now. 

So this is fine to get tests passing, and it's much better than than having ternaries everywhere, but I think it's worth considering a signature adjustment later.

Reply:
That seems reasonable to me - I think `xp=xp` fits better indeed. I didn't give it much thought here, since I focused on fixing the CI failures and the `eager_warns` signature was a reintroduction of what was previously reviewed.